frame/pools/src/lib.rs

```diff
@@ -0,0 +1,1135 @@
+
+			let unbonding_era = delegator.unbonding_era.ok_or(Error::<T>::NotUnbonding)?;
+			let current_era = T::StakingInterface::current_era().unwrap_or(Zero::zero());
+			if current_era.saturating_sub(unbonding_era) < T::StakingInterface::bonding_duration() {
+				return Err(Error::<T>::NotUnbondedYet.into())
+			};
+
+			let mut sub_pools =
+				SubPoolsStorage::<T>::get(&delegator.pool).ok_or(Error::<T>::SubPoolsNotFound)?;
+
+			let balance_to_unbond = if let Some(pool) = sub_pools.with_era.get_mut(&unbonding_era) {
+				let balance_to_unbond = pool.balance_to_unbond(delegator.points);
+				pool.points = pool.points.saturating_sub(delegator.points);
+				pool.balance = pool.balance.saturating_sub(balance_to_unbond);
+
+				balance_to_unbond
+			} else {
+				// A pool does not belong to this era, so it must have been merged to the era-less
+				// pool.
+				let balance_to_unbond = sub_pools.no_era.balance_to_unbond(delegator.points);
+				sub_pools.no_era.points = sub_pools.no_era.points.saturating_sub(delegator.points);
+				sub_pools.no_era.balance =
+					sub_pools.no_era.balance.saturating_sub(balance_to_unbond);
+
+				balance_to_unbond
+			};
+
+			T::StakingInterface::withdraw_unbonded(delegator.pool.clone(), num_slashing_spans)?;
+
+			T::Currency::transfer(
+				&delegator.pool,
+				&who,
+				balance_to_unbond,
+				ExistenceRequirement::AllowDeath,
+			)
+			.map_err(|e| {
+				log::warn!("system logic error: pool could not withdraw_unbonded due to lack of free balance");
+				e
+			})?;
+
+			SubPoolsStorage::<T>::insert(&delegator.pool, sub_pools);
+			Delegators::<T>::remove(&who);
+
+			Self::deposit_event(Event::<T>::Withdrawn {
+				delegator: who,
+				pool: delegator.pool,
+				amount: balance_to_unbond,
+			});
+
+			Ok(())
+		}
+
+		// Create a pool.
+		//
+		// * `targets`: _Stash_ addresses of the validators to nominate
+		// * `amount`: Balance to delegate to the pool. Must meet the minimum bond.
+		// * `index`: Disambiguation index for seeding account generation. Likely only useful when
+		//   creating multiple pools in the same extrinsic.
+		#[pallet::weight(666)]
+		pub fn create(
+			origin: OriginFor<T>,
+			targets: Vec<<T::Lookup as StaticLookup>::Source>,
+			amount: BalanceOf<T>,
+			index: u16,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(amount >= T::StakingInterface::minimum_bond(), Error::<T>::MinimumBondNotMet);
+
+			let (stash, reward_dest) = Self::create_accounts(index);
+
+			ensure!(!BondedPools::<T>::contains_key(&stash), Error::<T>::IdInUse);
+
+			T::StakingInterface::bond_checks(&stash, &stash, amount, &reward_dest)?;
+			let (stash, targets) = T::StakingInterface::nominate_checks(&stash, targets)?;
+
+			let mut bonded_pool = BondedPool::<T> { points: Zero::zero() };
+
+			// We must calculate the points to issue *before* we bond who's funds, else
+			// points:balance ratio will be wrong.
+			let points_to_issue = bonded_pool.points_to_issue(&stash, amount);
```

Comment:
You could also have a `fn issue(&mut self)` on the `BondedPool`.

Reply:
8c0feda9cd14999209d3bc5e19384b95c9e7a96e